Can't work with Storybook + Webpack 5 starting 1.13.0

Hi,

Sorry in advance for the vague issue because I couldn't build a repro and the error logs are hard to decipher, but here it is:
- I am using Node 18
- Which forces me to use Webpack 5 in Storybook, Webpack 4 won't work
- The new peer deps to "@testing-library/dom" and in particular "jsdom" raises issues related to some variables not being polyfilled

Samples of the error log:
```
ModuleNotFoundError: Module not found: Error: Can't resolve 'stream' in '/home/ubuntu/webapp/node_modules/jsdom/lib/jsdom/living/helpers'

BREAKING CHANGE: webpack < 5 used to include polyfills for node.js core modules by default.
This is no longer the case. Verify if you need this module and configure a polyfill for it.


ModuleNotFoundError: Module not found: Error: Can't resolve 'perf_hooks' in '/home/ubuntu/webapp/node_modules/jsdom/lib/jsdom'
    at /home/ubuntu/webapp/node_modules/webpack/lib/Compilation.js:2016:28
    at /home/ubuntu/webapp/node_modules/webpack/lib/NormalModuleFactory.js:798:13
    at eval (eval at create (/home/ubuntu/webapp/node_modules/webpack/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:10:1)
```

Any idea where to start?

--- a/.github/workflows/tests.yml
+++ b/.github/workflows/tests.yml
@@ -26,5 +26,5 @@ jobs:
           node-version: ${{ matrix.node-version }}
           cache: 'npm'
       - run: npm ci
-      - run: npm run prettier --check
+      - run: npm run check
       - run: npm run test:ci

--- a/.github/workflows/tests.yml
+++ b/.github/workflows/tests.yml
@@ -26,4 +26,5 @@ jobs:
           node-version: ${{ matrix.node-version }}
           cache: 'npm'
       - run: npm ci
+      - run: npm run prettier --check
       - run: npm run test:ci

--- a/__tests__/screenDebug.test.tsx
+++ b/__tests__/screenDebug.test.tsx
@@ -8,25 +8,25 @@ import {
 import { prettyShadowDOM, screen } from "../src/index";
 
 beforeEach(() => {
-	// jest.spyOn(console, 'log').mockImplementation(() => {})
-})
+  // jest.spyOn(console, 'log').mockImplementation(() => {})
+});
 
 afterEach(() => {
   // // @ts-expect-error
   // console.log.mockRestore()
-})
+});
 
 /* @see https://github.com/KonnorRogers/shadow-dom-testing-library/issues/33#issuecomment-1306593757 */
 test("Should not modify the original dom", () => {
-	render(<Duplicates />)
-	const originalHTML = document.body.innerHTML
-	expect(document.querySelector("shadow-root")).toBe(null)
+  render(<Duplicates />);
+  const originalHTML = document.body.innerHTML;
+  expect(document.querySelector("shadow-root")).toBe(null);
 
-	prettyShadowDOM(document.body)
+  prettyShadowDOM(document.body);
 
-	expect(document.querySelector("shadow-root")).toBe(null)
-	expect(originalHTML).toEqual(document.body.innerHTML)
-})
+  expect(document.querySelector("shadow-root")).toBe(null);
+  expect(originalHTML).toEqual(document.body.innerHTML);
+});
 
 test.skip("Triple shadow roots", async () => {
   render(<TripleShadowRoots />);

--- a/package.json
+++ b/package.json
@@ -14,6 +14,7 @@
   "scripts": {
     "test": "vitest run",
     "format": "npx prettier ./src ./__tests__ ./README.md --write",
+    "check": "npx prettier ./src ./__tests__ ./README.md --check",
     "test:watch": "vitest",
     "test:ci": "npm run build && npm run test && npm run jest",
     "jest": "jest",
@@ -46,8 +47,7 @@
   "url": "https://github.com/konnorrogers/shadow-dom-testing-library/issues",
   "repository": "github:konnorrogers/shadow-dom-testing-library",
   "peerDependencies": {
-    "@testing-library/dom": ">= 8",
-    "jsdom": ">= 20"
+    "@testing-library/dom": ">= 8"
   },
   "devDependencies": {
     "@open-wc/testing-helpers": "^2.1.2",

--- a/package.json
+++ b/package.json
@@ -51,8 +51,8 @@
   },
   "devDependencies": {
     "@open-wc/testing-helpers": "^2.1.2",
-    "@testing-library/jest-dom": "^5.16.4",
     "@testing-library/dom": ">= 8",
+    "@testing-library/jest-dom": "^5.16.4",
     "@testing-library/react": "^13.3.0",
     "@types/jest": "^28.1.4",
     "@types/node": "^18.0.3",
@@ -63,6 +63,7 @@
     "jest": "^28.1.2",
     "jest-environment-jsdom": "^28.1.2",
     "jest-preview": "^0.2.6",
+    "jsdom": "^20.0.2",
     "react": "^18.2.0",
     "react-dom": "^18.2.0",
     "standard-version": "^9.5.0",

--- a/package.json
+++ b/package.json
@@ -63,7 +63,6 @@
     "jest": "^28.1.2",
     "jest-environment-jsdom": "^28.1.2",
     "jest-preview": "^0.2.6",
-    "jsdom": ">= 20",
     "react": "^18.2.0",
     "react-dom": "^18.2.0",
     "standard-version": "^9.5.0",

--- a/src/index.ts
+++ b/src/index.ts
@@ -22,9 +22,9 @@ export * from "./types";
 export * from "./shadow-queries";
 
 export {
-	deepQuerySelector,
-	deepQuerySelectorAll
-} from "./deep-query-selectors"
+  deepQuerySelector,
+  deepQuerySelectorAll,
+} from "./deep-query-selectors";
 
 export {
   shadowScreen as screen,

--- a/src/jsdom-setup.ts
+++ b/src/jsdom-setup.ts
@@ -1,5 +1,0 @@
-import { TextEncoder, TextDecoder } from "util";
-
-/** This is a hack for using JSDOM + Jest. Vitest is unaffected by this. */
-global.TextEncoder = TextEncoder;
-global.TextDecoder = TextDecoder as typeof global["TextDecoder"];

--- a/src/pretty-shadow-dom.ts
+++ b/src/pretty-shadow-dom.ts
@@ -1,6 +1,4 @@
-import "./jsdom-setup";
 import { prettyDOM } from "@testing-library/dom";
-import { JSDOM } from "jsdom";
 
 /**
  * This is an extension of prettyDOM / logDOM that provides proper printing of shadow roots.
@@ -29,17 +27,18 @@ export function toJSDOM(element?: Element | Document | undefined): HTMLElement {
   // Remove extraneous whitespace as it creates bloated printing
   htmlString = htmlString.replace(/>\s+</g, "><");
 
-  const dom = new JSDOM(htmlString);
+  const parser = new DOMParser();
+  const dom = parser.parseFromString(htmlString, "text/html");
 
   if (
     element instanceof Document ||
     element instanceof HTMLHtmlElement ||
     element instanceof HTMLBodyElement
   ) {
-    return dom.window.document.body;
+    return dom.body;
   }
 
-  return dom.window.document.body.firstElementChild as HTMLElement;
+  return dom.body.firstElementChild as HTMLElement;
 }
 
 function processNodes(
@@ -61,7 +60,7 @@ function processNodes(
       const shadowRootPseudoNode = document.createElement("shadow-root");
       shadowRootPseudoNode.innerHTML = node.shadowRoot.innerHTML;
 
-			const clonedNode = node.cloneNode(true) as Element
+      const clonedNode = node.cloneNode(true) as Element;
       clonedNode.insertAdjacentElement("afterbegin", shadowRootPseudoNode);
 
       stringBuffer = stringBuffer.replace(outerHTML, clonedNode.outerHTML);

--- a/src/shadow-queries.ts
+++ b/src/shadow-queries.ts
@@ -107,11 +107,13 @@ const [
   getByShadowLabelText,
   findAllByShadowLabelText,
   findByShadowLabelText,
-] = toShadowQueries(buildQueries<ScreenShadowSelectorMatcherParams>(
-  queryAllByShadowLabelText,
-  getMultipleLabelTextError,
-  getMissingLabelTextError
-))
+] = toShadowQueries(
+  buildQueries<ScreenShadowSelectorMatcherParams>(
+    queryAllByShadowLabelText,
+    getMultipleLabelTextError,
+    getMissingLabelTextError
+  )
+);
 
 // Placeholder Text
 function queryAllByShadowPlaceholderText<T extends HTMLElement = HTMLElement>(

